app/authentication/authenticator.py

```diff
@@ -103,8 +108,10 @@ def load_user(extend_session: bool = True) -> Optional[User]:
         user_ik = cookie_session.get(USER_IK)
         user = User(user_id, user_ik)
 
-        if session_store.session_data and session_store.session_data.tx_id:
-            logger.bind(tx_id=session_store.session_data.tx_id)
+        metadata = get_metadata(user)
+
+        if metadata and metadata.get("tx_id"):
+            logger.bind(tx_id=metadata.get("tx_id"))
```

Comment:
I would do something similar to what you did with session data in langauge helpers, so we avoid getting it twice.
```suggestion
        if metadata and (tx_id := metadata.get("tx_id")):
            logger.bind(tx_id=tx_id)
```